### PR TITLE
ci: set up GitHub Actions. Add a test for x-smaxage

### PR DESCRIPTION
**Pull request by dunglas**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# Used as inspiration: https://github.com/mvdan/github-actions-golang
+
+name: Tests
+
+on:
+  push:
+  pull_request:
```
Comment (francislavoie) [inline .github/workflows/ci.yml:7]: I think it's better to specify the branch name, because otherwise it runs twice on PR pushes, iirc.

Comment (dunglas) [reply .github/workflows/ci.yml]: done

**Review by mholt — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      - master
+
+jobs:
+  test:
+    strategy:
+      # Default is true, cancels jobs for other platforms in the matrix if one fails
+      fail-fast: false
+      matrix:
+        # The build currently fails on Windows because of Olric. See https://github.com/buraksezer/olric/pull/64.
+        #os: [ ubuntu-latest, macos-latest, windows-latest ]
```
Comment (francislavoie) [inline .github/workflows/ci.yml:18]: It's been merged, should the dependency be updated to target a specific commit, and CI turned back on?

**Comment by francislavoie** [comment]
Ah, github.com/DataDog/zstd is a dependency which uses CGO, so that's a problem.

**Comment by dunglas** [comment]
@francislavoie according to https://github.com/golang/go/issues/40795 passing `-buildmode=exe` to `go build` should fix the problem, but it's not possible to pass this flag with `xcaddy` yet.

**Pull request merged**